iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/execution/config/executor/ClusterConfigTaskExecutor.java

```diff
@@ -1655,6 +1660,91 @@ public SettableFuture<ConfigTaskResult> createPipe(CreatePipeStatement createPip
     return future;
   }
 
+  @Override
+  public SettableFuture<ConfigTaskResult> alterPipe(AlterPipeStatement alterPipeStatement) {
+    SettableFuture<ConfigTaskResult> future = SettableFuture.create();
+
+    // Get pipe static meta
+    final String pipeName = alterPipeStatement.getPipeName();
+    final PipeStaticMeta pipeStaticMeta = PipeAgent.task().getPipeStaticMeta(pipeName);
+    if (Objects.isNull(pipeStaticMeta)) {
+      future.setException(
+          new IoTDBException(
+              String.format("Failed to alter pipe %s, the pipe does not exist", pipeName),
+              TSStatusCode.PIPE_ERROR.getStatusCode()));
+      return future;
+    }
+
+    // We do not support alter source plugin of pipe, so the previous configuration will be reused.
+    alterPipeStatement.setExtractorAttributes(
```

Comment:
I suggest pass the order directly to configNode here without checking logic other than the checks in PipePluginAgent.